main/src/ca/uwaterloo/flix/language/phase/TypeReconstruction.scala

```diff
@@ -251,15 +252,12 @@ object TypeReconstruction {
       val eff = Type.mkUnion(rs.map(_.exp.eff), loc)
       TypedAst.Expr.RestrictableChoose(star, e, rs, subst(tvar), eff, loc)
 
-    case KindedAst.Expr.ExtMatch(label, exp1, sym2, exp2, sym3, exp3, tvar, loc) =>
-      val e1 = visitExp(exp1)
-      val e2 = visitExp(exp2)
-      val e3 = visitExp(exp3)
-      val b2 = TypedAst.Binder(sym2, subst(sym2.tvar))
-      val b3 = TypedAst.Binder(sym3, subst(sym3.tvar))
-      val tpe = subst(tvar)
-      val eff = Type.mkUnion(e1.eff, e2.eff, e3.eff, loc)
-      TypedAst.Expr.ExtensibleMatch(label, e1, b2, e2, b3, e3, tpe, eff, loc)
+    case KindedAst.Expr.ExtMatch(exp, rules, loc) =>
+      val e = visitExp(exp)
+      val rs = rules.map(visitExtMatchRule)
+      val tpe = rs.head.exp.tpe
```

Comment:
Is this guaranteed to be nonempty?

Reply:
Sorry my bad. I was somewhere else in my mind :)

Reply:
I was also a bit 👀 but the other match does it too

Reply:
Yes the parsing rule for extensible match expects at least one rule